src/test/munit/directory-listener-functional-test-case.xml

```diff
@@ -0,0 +1,396 @@
+            </file:write>
+        </munit:behavior>
+        <munit:execution >
+            <try>
+                <until-successful maxRetries="10" millisBetweenRetries="1000" >
+                    <munit-tools:retrieve key="messages"/>
+                    <validation:validate-size value="#[payload]" min="1" max="1"/>
+                </until-successful>
+                <error-handler >
+                    <on-error-continue enableNotifications="false" logException="false" type="ANY">
+                        <munit-tools:retrieve key="messages"/>
+                    </on-error-continue>
+                </error-handler>
+            </try>
+            <set-variable value="#[payload]" variableName="files" />
+            <try>
+                <until-successful maxRetries="10" millisBetweenRetries="1000" >
+                    <munit-tools:retrieve key="messages"/>
+                    <validation:validate-size value="#[payload]" min="2"/>
+                </until-successful>
+                <munit-tools:fail message="There are more files than expected"/>
+                <error-handler>
+                    <on-error-continue enableNotifications="false" logException="false" type="MULE:RETRY_EXHAUSTED"/>
+                </error-handler>
+            </try>
+        </munit:execution>
+        <munit:validation>
+            <munit-tools:assert-that expression="#[vars.files[0].attributes.fileName]" is="#[MunitTools::equalTo('nonRecursive.txt')]"/>
+        </munit:validation>
+    </munit:test>
+
+    <flow name="source-move-to-test">
+        <file:listener config-ref="file-list-config" directory="matcherless" recursive="false" moveToDirectory="shared">
+            <scheduling-strategy >
+                <fixed-frequency/>
+            </scheduling-strategy>
+        </file:listener>
+        <flow-ref name="storeMessages"/>
+    </flow>
+    <munit:test name="move-to-test" description="Verifies that files are moved after processing">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="source-move-to-test"/>
+        </munit:enable-flow-sources>
+        <munit:behavior>
+            <file:write config-ref="file-list-config" path="matcherless/watchme.txt">
+                <file:content >who watches the watchmen?</file:content>
+            </file:write>
+        </munit:behavior>
+        <munit:execution >
+            <try >
+                <until-successful maxRetries="10" millisBetweenRetries="1000">
+                    <file:read config-ref="file-list-config" path="shared/watchme.txt"/>
+                    <validation:is-not-blank-string/>
+                </until-successful>
+                <set-variable value="#[payload]" variableName="file"/>
+                <try>
+                    <until-successful maxRetries="10" millisBetweenRetries="1000">
+                        <file:read config-ref="file-list-config" path="watchme.txt" />
```

Comment:
If this is about checking that the file does not exist anymore, you should check on "matcherless/watchme.txt" instead of "watchme.txt" since that is where it was created. As said on a previous comment, it would be nice if these paths were variables.

Reply:
You are right once again